packages/EasyStandard/src/Sniffs/Arrays/AlphabeticallySortedArrayKeysSniff.php

```diff
@@ -0,0 +1,260 @@
+<?php
+
+declare(strict_types=1);
+
+namespace EonX\EasyStandard\Sniffs\Arrays;
+
+use Error;
+use PHP_CodeSniffer\Files\File;
+use PHP_CodeSniffer\Sniffs\Sniff;
+use PhpParser\Comment;
+use PhpParser\Node\Expr\Array_;
+use PhpParser\Node\Expr\ArrayItem;
+use PhpParser\ParserFactory;
+use PhpParser\PrettyPrinter\Standard;
+use SlevomatCodingStandard\Helpers\TokenHelper;
+
+final class AlphabeticallySortedArrayKeysSniff implements Sniff
+{
+    /**
+     * @var string
+     */
+    public const ARRAY_KEYS_NOT_SORTED_ALPHABETICALLY = 'ArrayKeysNotSortedAlphabetically';
+
+    /**
+     * @var string
+     */
+    public const ARRAY_PROCESS_ERROR = 'ArrayProcessError';
+
+    /**
+     * @var string
+     */
+    public const FILE_PARSE_ERROR = 'FileParseError';
+
+    /**
+     * This comment is temporarily added to the lines of a multiline array to keep it multiline after fixing the order.
+     *
+     * @var string
+     */
+    private const TEMP_COMMENT_CONTENT = '[temp comment]';
+
+    /**
+     * A list of patterns to be checked to skip the array.
+     * Specify a token type (e.g. `T_FUNCTION` or `T_CLASS`) as a key
+     * and an array of regex patterns as a value to skip an array in the
+     * corresponding tokens (functions, classes).
+     *
+     * Example: `[T_FUNCTION => ['/^someFunction/']]`.
+     *
+     * @var mixed[]
+     */
+    public $skipPatterns = [];
+
+    /**
+     * @param int $bracketOpenerPointer
+     */
+    public function process(File $phpcsFile, $bracketOpenerPointer): void
+    {
+        if ($this->shouldSkip($phpcsFile, $bracketOpenerPointer)) {
+            return;
+        }
+
+        $tokens = $phpcsFile->getTokens();
+        $token = $tokens[$bracketOpenerPointer];
+        $bracketCloserPointer = $token['bracket_closer'] ?? $token['parenthesis_closer'];
+        $code = $phpcsFile->getTokensAsString($bracketOpenerPointer, $bracketCloserPointer - $bracketOpenerPointer + 1);
+
+        $parser = (new ParserFactory())->create(ParserFactory::PREFER_PHP7);
+        try {
+            $ast = $parser->parse('<?php' . PHP_EOL . $code . ';');
+        } catch (Error $error) {
+            $phpcsFile->addErrorOnLine(
+                "Parse error: {$error->getMessage()}",
+                $token['line'],
+                self::FILE_PARSE_ERROR
+            );
+
+            return;
+        }
+
+        if ($ast === null) {
+            $phpcsFile->addErrorOnLine(
+                'Unknown error while parsing the codee',
+                $token['line'],
+                self::FILE_PARSE_ERROR
+            );
+
+            return;
+        }
+
+        /** @var \PhpParser\Node\Stmt\Expression $stmtExpr */
+        $stmtExpr = $ast[0];
+        /** @var \PhpParser\Node\Expr\Array_ $array */
+        $array = $stmtExpr->expr;
+
+        $array = $this->refactor($array);
+
+        if ($array === null) {
```

Comment:
I think we can move that check before the previous expression and get rid of `null` check inside `refactor` method. Perhaps we can completely remove it if `$array` is `\PhpParser\Node\Expr\Array_` according to `@var` annotation.

Reply:
Let's rename `$array` to `$refactoredArray` after refactoring operation.

Reply:
Done.